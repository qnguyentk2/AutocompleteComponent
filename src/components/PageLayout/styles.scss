@import '../../../assets/styles/responsive.scss';

.container {
  background-image: url('../../../assets/imgs/background.jpg');
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  width: 100%;
  min-height: 800px;

  @include Responsive(SM) {
    background-position: 70% 50%;
  }
  
  @include Responsive(XS) {
    background-position: 70% 50%;
  }
  
  @include Responsive(XXS) {
    background-position: 70% 50%;
	}
}

.darkMode::after {
  content: ""; // ::before and ::after both require content
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.85);
  opacity: .7;
  z-index: 1;
}

.darkMode > div {
  z-index: 100 !important;
}

.darkMode > input {
  z-index: 100 !important;
}
