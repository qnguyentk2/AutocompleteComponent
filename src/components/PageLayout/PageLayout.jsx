import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Header from '@components/Header/Header';

import AppContent from '@components/AppContent/AppContent';
import styles from './styles.scss';

const PageLayout = props => {
  const {
    isLightMode,
    isBurgerMenuOpen,
  } = props;

  return (
    <div className={classnames(styles.container, isLightMode || isBurgerMenuOpen ? '' : styles.darkMode)}>
      <Header {...props} />
      <AppContent />
    </div>
  )
}

export default PageLayout;

PageLayout.propTypes = {
  isLightMode: PropTypes.bool,
  isBurgerMenuOpen: PropTypes.bool,
  // children: PropTypes.objectOf(PropTypes.any)
};

PageLayout.defaultProps = {
  isLightMode: true,
  isBurgerMenuOpen: false,
  // children: {}
};