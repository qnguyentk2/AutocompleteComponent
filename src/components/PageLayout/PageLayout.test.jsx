/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import toJSON from 'enzyme-to-json';
import PageLayout from './PageLayout';

const propsData = {
  isLightMode: true,
  isBurgerMenuOpen: true
}

const propsData1 = {
  isLightMode: false,
  isBurgerMenuOpen: false
}


const setup = props => {
  const wrapper = mount(
    <MemoryRouter>
      <PageLayout {...props} />
    </MemoryRouter>
  );

  return { wrapper };
}

describe('Test PageLayout', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup(propsData);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('renders without crashing with dark mode', () => {
    const { wrapper } = setup(propsData1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
})