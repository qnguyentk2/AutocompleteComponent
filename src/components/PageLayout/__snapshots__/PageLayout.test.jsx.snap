// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PageLayout renders without crashing 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "5dmvm4",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "5dmvm4",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PageLayout
      isBurgerMenuOpen={true}
      isLightMode={true}
    >
      <div
        className="container"
      >
        <withRouter(Header)
          isBurgerMenuOpen={true}
          isLightMode={true}
        >
          <Header
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "5dmvm4",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "5dmvm4",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isBurgerMenuOpen={true}
            isLightMode={true}
            location={
              Object {
                "hash": "",
                "key": "5dmvm4",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="Header"
            >
              <div>
                <Link
                  to="/"
                >
                  <LinkAnchor
                    href="/"
                    navigate={[Function]}
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <img
                        alt="logo"
                        src={Object {}}
                      />
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
              <div
                className="leftSide"
              >
                <a
                  className="themeMode"
                >
                  <img
                    alt="theme"
                    src={Object {}}
                  />
                </a>
                <Link
                  key="Become a host"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Become a host
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  key="Help"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Help
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  key="Sign up"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Sign up
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  key="Log in"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Log in
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </Header>
        </withRouter(Header)>
        <AppContent>
          <withRouter(Routes)>
            <Routes
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "5dmvm4",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "5dmvm4",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "5dmvm4",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Switch>
                <Route
                  component={[Function]}
                  computedMatch={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  exact={true}
                  key="/"
                  location={
                    Object {
                      "hash": "",
                      "key": "5dmvm4",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  path="/"
                >
                  <Autocomplete
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "canGo": [Function],
                        "createHref": [Function],
                        "entries": Array [
                          Object {
                            "hash": "",
                            "key": "5dmvm4",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                        ],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "index": 0,
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "key": "5dmvm4",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "key": "5dmvm4",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <div
                      className="Autocomplete darkMode"
                    >
                      <img
                        alt="search"
                        src={Object {}}
                      />
                      <input
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        type="text"
                        value=""
                      />
                    </div>
                  </Autocomplete>
                </Route>
              </Switch>
            </Routes>
          </withRouter(Routes)>
        </AppContent>
      </div>
    </PageLayout>
  </Router>
</MemoryRouter>
`;

exports[`Test PageLayout renders without crashing with dark mode 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "cyxjsz",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "cyxjsz",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PageLayout
      isBurgerMenuOpen={false}
      isLightMode={false}
    >
      <div
        className="container darkMode"
      >
        <withRouter(Header)
          isBurgerMenuOpen={false}
          isLightMode={false}
        >
          <Header
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "cyxjsz",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "cyxjsz",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isBurgerMenuOpen={false}
            isLightMode={false}
            location={
              Object {
                "hash": "",
                "key": "cyxjsz",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="Header"
            >
              <div>
                <Link
                  to="/"
                >
                  <LinkAnchor
                    href="/"
                    navigate={[Function]}
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <img
                        alt="logo"
                        src={Object {}}
                      />
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
              <div
                className="leftSide"
              >
                <a
                  className="themeMode"
                >
                  <img
                    alt="theme"
                    src={Object {}}
                  />
                </a>
                <Link
                  key="Become a host"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Become a host
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  key="Help"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Help
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  key="Sign up"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Sign up
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  key="Log in"
                  to="/become-a-host"
                >
                  <LinkAnchor
                    href="/become-a-host"
                    navigate={[Function]}
                  >
                    <a
                      href="/become-a-host"
                      onClick={[Function]}
                    >
                      Log in
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </Header>
        </withRouter(Header)>
        <AppContent>
          <withRouter(Routes)>
            <Routes
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "cyxjsz",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "cyxjsz",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "cyxjsz",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Switch>
                <Route
                  component={[Function]}
                  computedMatch={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  exact={true}
                  key="/"
                  location={
                    Object {
                      "hash": "",
                      "key": "cyxjsz",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  path="/"
                >
                  <Autocomplete
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "canGo": [Function],
                        "createHref": [Function],
                        "entries": Array [
                          Object {
                            "hash": "",
                            "key": "cyxjsz",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                        ],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "index": 0,
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "key": "cyxjsz",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "key": "cyxjsz",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <div
                      className="Autocomplete darkMode"
                    >
                      <img
                        alt="search"
                        src={Object {}}
                      />
                      <input
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        type="text"
                        value=""
                      />
                    </div>
                  </Autocomplete>
                </Route>
              </Switch>
            </Routes>
          </withRouter(Routes)>
        </AppContent>
      </div>
    </PageLayout>
  </Router>
</MemoryRouter>
`;
